pkg/operator/apis/monitoring/v1alpha1/types_logs.go

```diff
@@ -365,7 +363,16 @@ type MatchStageSpec struct {
 
 	// Nested set of pipeline stages to execute when action: keep and the log
 	// line matches selector.
-	Stages []*PipelineStageSpec `json:"stages,omitempty"`
+	//
+	// An example value for stages may be:
+	//
+	//   stages: |
+	//     - json: {}
+	//     - labelAllow: [foo, bar]
+	//
+	// Note that stages is a string and because SIG API Machinery does not support
+	// recursive types. Be careful not to mistype anything.
+	Stages string `json:"stages,omitempty"`
```

Comment:
I tried a few different ways to avoid this, but this really seems to be the only sane way to do it without creating _more_ CRDs.